### PR TITLE
Add Eclipse formatter and checks

### LINKED ISSUE #31: Set up custom formatter and compiler checks
Can be copied from https://github.com/hbz/lobid-organisations/tree/master/.settings

See http://hbz.github.io/#dev-process-coding-conventions

### PR DESCRIPTION
**Pull request by SBRitter**
Summary:
- Add Eclipse formatter and compiler (same as lobid-organisations)
- Refactor code according to compiler warnings

Will resolve #31 

Note: for some reason, two commits by philboeselager from issue #25 that are already merged in the master (pull request: #33) are contained in this pull request, too. The commits corresponding to the two in the master are these: 3d858762046f7c33e7a48b12c7693d4ea8996a11 and 8107e506e411095acfa14c5d3ff6579b080c64a8

### CODE REVIEW

File: .settings/.settings/org.eclipse.m2e.core.prefs [hunk 1]
```diff
@@ -0,0 +1,4 @@
+activeProfiles=
+eclipse.preferences.version=1
+resolveWorkspaceProjects=true
+version=1
```
Comment (fsteeg) [inline .settings/.settings/org.eclipse.m2e.core.prefs]: I think `org.eclipse.m2e.core.prefs` can be removed, it's no longer a Maven project.

File: .settings/.settings/org.scala-ide.sdt.core.prefs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#Generated by sbteclipse
+#Mon Jul 04 19:04:02 CEST 2016
+scala.compiler.additionalParams=-encoding utf8
+deprecation=true
+unchecked=true
+scala.compiler.useProjectSettings=true
```
Comment (fsteeg) [inline .settings/.settings/org.scala-ide.sdt.core.prefs]: We should .gitignore `org.scala-ide.sdt.core.prefs`, which is generated by sbteclipse.

File: .settings/.settings/org.eclipse.core.resources.prefs [hunk 3]
```diff
@@ -0,0 +1,3 @@
+eclipse.preferences.version=1
+encoding//test/transformation/input/dbs.csv=ISO-8859-1
+encoding/<project>=UTF-8
```
Comment (fsteeg) [inline .settings/.settings/org.eclipse.core.resources.prefs]: This looks like a duplicate settings file (same for all in `.settings/.settings`).

File: .settings/org.eclipse.m2e.core.prefs [hunk 4]
```diff
@@ -0,0 +1,4 @@
+activeProfiles=
+eclipse.preferences.version=1
+resolveWorkspaceProjects=true
+version=1
```
Comment (fsteeg) [inline .settings/org.eclipse.m2e.core.prefs]: I think `org.eclipse.m2e.core.prefs` can be removed, it's no longer a Maven project.

File: .settings/org.scala-ide.play2.prefs [hunk 5]
```diff
@@ -0,0 +1,4 @@
+#Generated by sbt-plugin
+#Fri Jul 08 14:17:49 CEST 2016
+eclipse.preferences.version=1
+templateImports=import play.api.templates._\nimport play.api.templates.PlayMagic._\nimport models._\nimport controllers._\nimport java.lang._\nimport java.util._\nimport scala.collection.JavaConversions._\nimport scala.collection.JavaConverters._\nimport play.api.i18n._\nimport play.core.j.PlayMagicForJava._\nimport play.mvc._\nimport play.data._\nimport play.api.data.Field\nimport play.mvc.Http.Context.Implicit._\nimport views.%format%._\n
```
Comment (fsteeg) [inline .settings/org.scala-ide.play2.prefs]: We should .gitignore `org.scala-ide.play2.prefs`, which is generated by sbteclipse.

File: .settings/org.scala-ide.sdt.core.prefs [hunk 6]
```diff
@@ -0,0 +1,6 @@
+#Generated by sbteclipse
+#Mon Jul 04 19:04:02 CEST 2016
+scala.compiler.additionalParams=-encoding utf8
+deprecation=true
+unchecked=true
+scala.compiler.useProjectSettings=true
```
Comment (fsteeg) [inline .settings/org.scala-ide.sdt.core.prefs]: We should .gitignore `org.scala-ide.sdt.core.prefs`, which is generated by sbteclipse.

File: app/controllers/Application.java [hunk 7]
```diff
@@ -3,10 +3,12 @@
 import play.mvc.Controller;
 import play.mvc.Result;
 
+@SuppressWarnings("javadoc")
```
Comment (fsteeg) [inline app/controllers/Application.java]: Ignoring missing Javadocs is fine for test classes, but for production code the reason for the custom compiler check is to actually ensure all public API is documented, so we should add documentation here.

File: app/controllers/Global.java [hunk 8]
```diff
@@ -7,6 +7,7 @@
 import play.GlobalSettings;
 import play.Logger;
 
+@SuppressWarnings("javadoc")
```
Comment (fsteeg) [inline app/controllers/Global.java]: Ignoring missing Javadocs is fine for test classes, but for production code the reason for the custom compiler check is to actually ensure all public API is documented, so we should add documentation here.

File: app/controllers/geo/Constants.java [hunk 9]
```diff
@@ -1,5 +1,6 @@
 package controllers.geo;
 
+@SuppressWarnings("javadoc")
```
Comment (fsteeg) [inline app/controllers/geo/Constants.java]: Ignoring missing Javadocs is fine for test classes, but for production code the reason for the custom compiler check is to actually ensure all public API is documented, so we should add documentation here.

File: app/controllers/geo/GeoElasticsearch.java [hunk 10]
```diff
@@ -12,67 +12,89 @@
...
 import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.node.Node;
 
 import com.typesafe.config.Config;
 import com.typesafe.config.ConfigFactory;
 
+/**
+ * Configuration and creation of Elasticsearch index
+ */
 @SuppressWarnings("javadoc")
```
Comment (fsteeg) [inline app/controllers/geo/GeoElasticsearch.java]: Ignoring missing Javadocs is fine for test classes, but for production code the reason for the custom compiler check is to actually ensure all public API is documented, so we should add documentation here.

File: app/controllers/geo/GeoInformator.java [hunk 11]
```diff
@@ -11,6 +11,11 @@
 import play.mvc.Controller;
 import play.mvc.Result;
 
+/**
+ * Methods to trigger geo lookups on the index
+ *
+ */
+@SuppressWarnings("javadoc")
```
Comment (fsteeg) [inline app/controllers/geo/GeoInformator.java]: Ignoring missing Javadocs is fine for test classes, but for production code the reason for the custom compiler check is to actually ensure all public API is documented, so we should add documentation here.

File: app/controllers/geo/QueryHelpers.java [hunk 12]
```diff
@@ -13,54 +13,61 @@
 import org.json.JSONException;
 import org.json.JSONObject;
 
+/**
+ * Helper methods for queries
+ */
+@SuppressWarnings("javadoc")
```
Comment (fsteeg) [inline app/controllers/geo/QueryHelpers.java]: Ignoring missing Javadocs is fine for test classes, but for production code the reason for the custom compiler check is to actually ensure all public API is documented, so we should add documentation here.

File: app/controllers/geo/Constants.java (hunk 9 above)
Comment (SBRitter) [inline app/controllers/geo/Constants.java]: I agree! However, these constants are really self-explanatory, aren't they? That is, I can add the comments, but will it really add information?

**Comment by fsteeg** [comment]
+1

**Pull request merged**